core/src/main/java/org/bitcoinj/core/LegacyAddress.java

```diff
@@ -63,13 +63,24 @@ public class LegacyAddress extends Address {
      *            20-byte hash of pubkey or script
      */
     private LegacyAddress(Network network, boolean p2sh, byte[] hash160) throws AddressFormatException {
-        super(network, hash160);
+        super(normalizeNetwork(network), hash160);
         if (hash160.length != 20)
             throw new AddressFormatException.InvalidDataLength(
                     "Legacy addresses are 20 byte (160 bit) hashes, but got: " + hash160.length);
         this.p2sh = p2sh;
     }
 
+    // LegacyAddress does not distinguish between the different testnet types, normalize to TESTNET
```

Comment:
I think the comment should be inside the method, just before the `if`.

Reply:
Will fix.